lib/Client.php

```diff
@@ -90,6 +90,35 @@ public function __construct(
 		}
 	}
 
+	public function requestUserInfo($attribute = null) {
+		if (true) {
+			$resp = $this->requestTokenExchange(
+				$this->getRefreshToken(),
+				'urn:ietf:params:oauth:token-type:refresh_token',
+				$this->getClientID()
+			);
+
+			$accessToken = $resp->access_token;
+			$userInfo = $this->introspectToken($accessToken);
```

Comment:
Just spotted this: this is anyhow wrong.

Request user info endpoint will only be for requesting the user info endpoint.

The logic to use either one or the other is handled in the app itself.

Reply:
Where should this logic go to then? From what I gather requestUserInfo is always called. We could optimize this request away if we get the userinfo from the id-token....

Reply:
I guess this is a special case that /userinfo is disabled.